### ISSUE TITLE
Support for "array of string" in `approved-reviews-by`

### ISSUE THREAD

**Issue by SteVwonder** [Mergifyio/mergify-engine#656]
In this PR (https://github.com/flux-framework/docs/pull/9), I am trying to have multiple usernames/teams that can approve PRs for mergify.  Based on [this documentation](https://doc.mergify.io/conditions.html#attribute-list), it says the `approved-reviews-by` can be an "array of string".  I can't seem to get that to work.  Both `- approved-reviews-by=[@flux-framework/core,@flux-framework/docs]` and

```yaml
- approved-reviews-by=
  - @flux-framework/core 
  - @flux-framework/docs
```

are rejected.

After poking around in the rules parser, [this line](https://github.com/Mergifyio/mergify-engine/blob/4d8f774d801355eb5aca6a8791d3b16367cdfab8/mergify_engine/rules/parser.py#L21) makes it look like that config option only supports a single username/team.  Is that right?

**Comment by jd** [comment]
The documentation says it *is* an array of string.
There's no `or` operator in the conditions, so you'd need to duplicate your rules, but you can do something like:


```
pull_request_rules:
  - name: merge for team docs
    conditions:
       - approved-reviews-by=@flux-framework/docs
    actions:
       merge:
          method: merge

  - name: merge for team core
    conditions:
       - approved-reviews-by=@flux-framework/core
    actions:
       merge:
          method: merge
```

**Comment by SteVwonder** [comment]
> There's no or operator in the conditions, so you'd need to duplicate your rules, but you can do something like:

Thanks! That seems like it will work well.  One quick clarifying question (I did look in the docs, but pre-apologizes if I missed the answer):  what happens if both rules are simultaneously satisfied?  For example, a user that has membership in both groups approves a PR (or even two members, each from one group approve).  In our case, the actions for both rules are the same, so I'm not worried about different actions being performed.  I just want to double check that the merge action isn't performed twice somehow.

> The documentation says it is an array of string.  If you have any suggestion I'd be happy to make the doc clearer :)

I think most of my confusion was around `array`.  When I think of `array of string` in the context of YAML, I think of either of the examples that I posted above (both the short and long forms of arrays in YAML).  I may just not fully understand the config spec, but from what I can tell, I would say that the `approved-review-by` value is just a `string`, not an `array of strings`.
